content/articles/how-to-ship-zero-javascript-with-astro/index.md

```diff
@@ -0,0 +1,464 @@
+ 
+* A command line or terminal(Nodejs 14.15.1 or higher).
+ 
+* A code editor(IDE). I use [VS Code](https://code.visualstudio.com/download).
+ 
+* Google Chrome.
+ 
+### Installations
+ 
+You need to install Node.js version 14.15.1 or higher. Visit [Node.js](https://nodejs.org/en/) official site to download and install the latest version of Node.js. It comes with a pre-installed node package manager(NPM).
+ 
+To check  Node.js is installation and the version installed. Open your terminal and type in the command below:
+ 
+```bash
+ 
+ node --version
+ 
+```
+ 
+To verify the installation and version of NPM type in the command below in your terminal.
+ 
+```bash
+ 
+npm --version
+ 
+```
+ 
+### Step 1 - Creating a New Astro Project
+ 
+Create a new directory, The directory will contain all your installed dependencies.
+ 
+Navigate to your command line and type in the line below. You can use the integrated terminal in your code editor.
+ 
+```bash
+ 
+mkdir astro-app
+ 
+```
+ 
+Next, navigate to the directory using the command below
+ 
+```bash
+ 
+cd astro-app
+ 
+```
+ 
+In your terminal type in the below to initialize your Astro project:
+ 
+```bash
+ 
+npm init astro
+ 
+```
+ 
+The command above will prompt a message, which asks for the template to start with. For this article, you will select **Starter Kit(Generic)**.
+ 
+![astro-starting-template](/engineering-education/how-to-ship-zero-javascript-with-astro/astro-starting-template.jpg)
+ 
+The above selection will prompt another message. This message will ask the framework you will like to use as shown in the image below. Select React for this article.
+ 
+![select-framwork](/engineering-education/how-to-ship-zero-javascript-with-astro/select-framework.png)
+ 
+Astro will then copy some project files to start our project.
+ 
+Before we go further. Type in the command below in the terminal. This command installs the required packages(dependencies):
+ 
+```bash
+ 
+npm install
+ 
+```
+ 
+### Step 2 - Application Structure
+ 
+In this step, we will have an overview of what our application folder structure looks like.
+ 
+We have different folders such as:
+ 
+* src
+ 
+* components
+ 
+* layouts
+ 
+* pages
+ 
+* public
+ 
+* package.json
+ 
+The ones we are more interested in are:
+ 
+* **src**: It contains the components, layout, and pages folder. The src is where our project source codes will be.
+ 
+* **components**: This is inside the src folder. The components folder will contain our React UI components.
+ 
+* **pages**: This is inside the src folder. Pages folder includes the **index.astro**, this serves as our entry point
+ 
+* **astro.config.mjs**: This is a configuration file for our Astro project. It contains a configuration object.
+ 
+```
+ 
+├── src/
+ 
+│   ├── components/
+ 
+|   |       |── Tours.astro
+ 
+│   ├── layouts/
+ 
+│   └── pages/
+ 
+│       └── index.astro
+ 
+├── public/
+ 
+├── astro.config.mjs
+ 
+└── package.json
+ 
+```
```

Comment:
I followed your steps and the generated boilerplate is like this:

```
/
├── public/
│   ├── robots.txt
│   └── favicon.ico
├── src/
│   ├── components/
│   │   └── Tour.astro
│   └── pages/
│       └── index.astro
└── package.json
```
Make sure to on that it does change the import paths.